test/unit/streaming.controllers.MediaController.js

```diff
@@ -450,6 +471,114 @@ describe('MediaController', function () {
+        });
+
+        it('should check initial media settings to choose initial track with a lang and empty accessibility setting', function () {
+            mediaController.addTrack(enTrack);
+            mediaController.addTrack(enADTrack);
+            mediaController.addTrack(esTrack);
+
+            let trackList = mediaController.getTracksFor(trackType, streamInfo.id);
+            expect(trackList).to.have.lengthOf(3);
+            expect(objectUtils.areEqual(trackList[0], enTrack)).to.be.true; // jshint ignore:line
+            expect(objectUtils.areEqual(trackList[1], enADTrack)).to.be.true; // jshint ignore:line
+            expect(objectUtils.areEqual(trackList[2], esTrack)).to.be.true; // jshint ignore:line
+
+            let currentTrack = mediaController.getCurrentTrackFor(trackType, streamInfo.id);
+            expect(objectUtils.areEqual(currentTrack, enTrack)).to.be.false; // jshint ignore:line
+            expect(objectUtils.areEqual(currentTrack, enADTrack)).to.be.false; // jshint ignore:line
+            expect(objectUtils.areEqual(currentTrack, esTrack)).to.be.false; // jshint ignore:line
+
+            // call to setInitialMediaSettingsForType
+            mediaController.setInitialSettings(trackType, {
+                lang: 'en',
+                accessibility: ''
+            });
+            mediaController.setInitialMediaSettingsForType(trackType, streamInfo);
+
+            currentTrack = mediaController.getCurrentTrackFor(trackType, streamInfo.id);
+            expect(objectUtils.areEqual(currentTrack, enTrack)).to.be.true; // jshint ignore:line
+        });
+
+        it('should check initial media settings to choose initial track with a lang and accessibility', function () {
+            mediaController.addTrack(enTrack);
+            mediaController.addTrack(enADTrack);
+            mediaController.addTrack(esTrack);
+
+            let trackList = mediaController.getTracksFor(trackType, streamInfo.id);
+            expect(trackList).to.have.lengthOf(3);
+            expect(objectUtils.areEqual(trackList[0], enTrack)).to.be.true; // jshint ignore:line
+            expect(objectUtils.areEqual(trackList[1], enADTrack)).to.be.true; // jshint ignore:line
+            expect(objectUtils.areEqual(trackList[2], esTrack)).to.be.true; // jshint ignore:line
+
+            let currentTrack = mediaController.getCurrentTrackFor(trackType, streamInfo.id);
+            expect(objectUtils.areEqual(currentTrack, enTrack)).to.be.false; // jshint ignore:line
+            expect(objectUtils.areEqual(currentTrack, enADTrack)).to.be.false; // jshint ignore:line
+            expect(objectUtils.areEqual(currentTrack, esTrack)).to.be.false; // jshint ignore:line
+
+            // call to setInitialMediaSettingsForType
+            mediaController.setInitialSettings(trackType, {
+                lang: 'en',
+                accessibility: 'description'
```

Comment:
Should this match the accessiblity defined in the track :`1`?. Otherwise two tracks are returned from the filter function and the track with the highest selectionPriority is selected. Ideally this test also assigns a lower selection priority to the track with the `accessiblity` parameter to confirm that the filter works

Reply:
Sorry, I have to admit that while rebasing, I made quite some mistakes (fixed now).
The interim solution now is to keep changes from both branches for best test coverage.

I may come up with changes to combine test cases to reduce execution time...